hoot-core-test/src/test/cpp/hoot/core/criterion/BuildingCriterionTest.cpp

```diff
@@ -59,10 +57,10 @@ class BuildingCriterionTest : public HootTestFixture
     HOOT_STR_EQUALS(0, uut.isSatisfied(TestUtils::getElementWithNote(map, "bestbuy")));
     HOOT_STR_EQUALS(0, uut.isSatisfied(TestUtils::getElementWithNote(map, "pho")));
     HOOT_STR_EQUALS(0, uut.isSatisfied(TestUtils::getElementWithNote(map, "panera")));
-    HOOT_STR_EQUALS(1, uut.isSatisfied(TestUtils::getElementWithNote(map, "freddys")));
-    HOOT_STR_EQUALS(1, uut.isSatisfied(TestUtils::getElementWithNote(map, "jewelry")));
+    HOOT_STR_EQUALS(0, uut.isSatisfied(TestUtils::getElementWithNote(map, "freddys")));
+    HOOT_STR_EQUALS(0, uut.isSatisfied(TestUtils::getElementWithNote(map, "jewelry")));
     HOOT_STR_EQUALS(1, uut.isSatisfied(TestUtils::getElementWithNote(map, "paneragroup")));
-    HOOT_STR_EQUALS(0, uut.isSatisfied(TestUtils::getElementWithNote(map, "jewelryandfreddys")));
+    HOOT_STR_EQUALS(1, uut.isSatisfied(TestUtils::getElementWithNote(map, "jewelryandfreddys")));
   }
```

Comment:
Since shop=mall is now in the building category, any feature that has that tag but doesn't have building=yes will be called a building now. It seems unlikely that you would real data with shop=mall and w/o building=yes.